### ISSUE TITLE
Migrate or join conda-recipe efforts in conda-forge?

### ISSUE THREAD

**Issue by h-vetinari** [oxfordcontrol/osqp-python#32]
Hey everyone

I've been working with cvxpy and osqp for a while, and it's always been a pain to install (especially on windows). Things have progressed a lot over the last two years, but only recently did I become hopeful to have a centralised install of everything necessary through conda-forge.

The reason a single channel is important for several reasons:
* conda-forge is the "gold standard" of curated packages in the conda-universe, and even anaconda uses this as their "upstream" for most packages
* in turn, it is strongly discouraged to add other channels, which might have packages compiled with incompatible compilers etc.
* consequently, it is often hard to get any channel other than conda-forge (and main) past a sys-admin / firewall / proxy.
* finally, it gives much more exposure. Although the current [osqp-feedstock](https://github.com/conda-forge/osqp-feedstock) has seen no action since more than half a year (and didn't have a windows build), the downloads outnumber the oxfordcontrol-channel by more than [four](https://anaconda.org/conda-forge/osqp) to [one](https://anaconda.org/oxfordcontrol/osqp).

After I discovered that conda-forge/osqp-feedstock#8 had been dormant for a while, I managed to build for [0.5.0](https://github.com/conda-forge/osqp-feedstock/pull/9) a couple of days ago, and consequently also for [windows](https://github.com/conda-forge/osqp-feedstock/pull/10). Note that this involved cutting some corners (e.g. I really don't care about investing any time in python 2.7), and some tiny patches to get stuff to build on azure.

Now I've run into problems with [0.6.0](https://github.com/conda-forge/osqp-feedstock/pull/11) and [0.6.1](https://github.com/conda-forge/osqp-feedstock/pull/12), and was trying to figure out where the specification of the debug-build came from, I saw that there was a conda-recipe in this repo as well.

Long story short: I wanted to **suggest** merging forces and to (for the above reasons and more) concentrate the conda-related efforts on the conda-forge feedstock. I'm happy to add you to the maintainers right away, and don't care whether we revert/adapt any and all changes I made. All I want is a working conda installation across all platforms and with minimal maintenance effort for everyone involved.

**Comment by bstellato** [comment]
I apologize cvxpy and osqp were a pain for you but on our side we made a huge effort making osqp always easy to install with just one command from any language.

Thanks a lot for the efforts in the osqp-feedstock. The only reasons why we continued supporting our own channel were
- the code generation checks were not enabled in the osqp-feedstock (but I see that you enabled them recently)
- the ability to create, package and deploy releases automatically from git tags in this repository.

I would be happy to migrate our conda packaging to conda-forge. However, I am not extremely familiar with the feedstock system and what is required to make new releases. Is there any smoother way to create releases to conda-forge rather than opening a new PR and making sure to tick all the boxes? I am just thinking in terms of convenience since we are trying to support several languages and it is quite tedious to do the packaging each time for every language.

In any case, I would be happy with the transition. Please add @gbanjac and me to the conda-feedstock maintainers so that we can start working on it.

**Comment by h-vetinari** [comment]
@bstellato 
Please don't apologize, I really appreciate all the work that has gone into making this easier already!

> However, I am not extremely familiar with the feedstock system and what is required to make new releases.

It is very easy - every merged pull request (that passes the building CI) will generate a new package on conda-forge. It is therefore important that each build gets a new build number (defined in `meta.yaml`), if the version does not change. But the pull request template will always remind you of that anyway.

Also, there's a bot that will open Pull requests if a new version is published to the upstream that the package is taken from (GH or PyPI), and usually that just means changing the hashes.

The one thing of your current workflow that's not automatically supported is building packages for every single commit of this repository. But it is (should be) possible to build packages for alphas, betas and rcs, that get "hidden" from normal users behind a `--pre` CLI-argument.

I already added you to the maintainers, hope this works out. Glad to have you on board!

Note that conda-forge may use different compilers than you do (hence e.g. I had to change to `NMake CMakelists`, see [here](https://github.com/conda-forge/osqp-feedstock/pull/10#issuecomment-525947008)). There are a lot of `@conda-forge/help-[...]` lists of people who should be able to help you with any problems though. I'll be gone over the weekend, but I'll also try to help of course. :)

**Comment by bstellato** [comment]
This is fixed now. We are using conda-forge for the conda package now. https://github.com/conda-forge/osqp-feedstock

This is also the recommended installation procedure in the [OSQP docs](https://osqp.org/docs/get_started/python.html)
